### ISSUE TITLE
Priority Research

### ISSUE THREAD

**Issue by cammy1o** [ldtteam/minecolonies-features#791]
Button Text: "Add to Priority Queue"

Description/Tooltip: "Click to prioritize this research task and move it to the front of the queue."

When a user clicks on this button for a particular research task, it could move that task to the top of the list or queue of research tasks that need to be completed. This would ensure that it gets researched first before other tasks.

**Comment by Talyda** [comment]
Would that actually be useful at all?  A researcher can only work on one research at a time, unless you have more levels in your university than you have researchers, and then I think that is the price you pay for not fully staffing your uni?
Like you can't have 5 researchers all working on one research, so it wouldn't get done any faster

**Comment by Thodor12** [comment]
Every researcher randomly selects a research to work on, and ticks that one down. It's not a 1:1 researcher-research

**Comment by Talyda** [comment]
But it's still only one researcher per research?
So this wouldn't really do anything would it? Unless you had too few researchers for researches, and I thought that was a deliberate choice?

**Comment by Talyda** [comment]
What I mean is, if you have a level 4 uni and 4 researchers, it won't make any difference.
If you have a level 4 uni, and 3 researchers, that is the price you pay?
(sorry, not sure my words are clear =P)

**Comment by Thodor12** [comment]
Even then, every researcher randomly selects what research to progress. There's no distribution that researchers evenly spread out.
The suggestion is to add a prioritization so that that given research gets done first, instead of everyone picking a random thing everytime

**Comment by Talyda** [comment]
Yeah, I guess I just don't understand what difference it would make in the long run. Your researchers will still be working on it, regardless of who picks it first in the morning?

**Comment by Thodor12** [comment]
No, you lack some implementation context, every tick that they will perform a "study" operation they select a new research again and again.
This way, in the worst case, they can constantly be doing research on a very long - not so important - research instead of what you actually need

**Comment by Talyda** [comment]
So all 5 could be working on the same one thing?

**Comment by Thodor12** [comment]
Exactly, there is nothing in place that prevents them from all working on the same thing

**Comment by MotionlessTrain** [comment]
^ How does it prevent that when you have a level 5 university with 5 researches and only 1 researcher going on, that that research is going 5 times as fast?
That is what could happen with the prioritised one too, and the researches are not really going in parallel
(I haven't noticed any speed difference between having only 1 researcher working on a level 1 research, and having 2 researchers, and only 1 level 1 research going on)

**Comment by Thodor12** [comment]
The problem is, there's no uniform control over it, there's no "evenly spread across research", "one researcher to one research", it's just a pure random, which makes for a bad indicator of what's going on.
We either have to ensure it's not a pure random, or ensure we have some way for players to actually tell them what to do.

**Comment by Talyda** [comment]
Oh! Then that sounds like a different problem entirely.
Nothing to do with prioritising a research. 
Actually, if that is true, if we had 5 researchers and only one research, should that one not be.. yeah, what train said.
They need to each be doing a separate research.

**Comment by Thodor12** [comment]
Can always make that a research on itself to make the researchers smarter and work more efficiently 😅
